packages/boilerplates/ICanFindOnlineBoilerplatePackages.ts

```diff
@@ -0,0 +1,45 @@
+/*---------------------------------------------------------------------------------------------
+*  Copyright (c) Dolittle. All rights reserved.
+*  Licensed under the MIT License. See LICENSE in the project root for license information.
+*--------------------------------------------------------------------------------------------*/
+
+import { BoilerplatePackageJson } from "./internal";
+
+/**
+ * Responsible for discovering boilerplates
+ *
+ * @export
+ * @interface ICanFindOnlineBoilerplatePackages
+ */
+export interface ICanFindOnlineBoilerplatePackages {
+    
+    /**
+     * Finds boilerplate packages online and and returns the package.json objects of the latest versions of the boilerplates packages compatible with the common tooling major version
+     *
+     * @param {string[]} [keywords] Additional keywords used in search
+     * @param {number} [limit] The limit of boilerplates
+     * @returns {Promise<BoilerplatePackageJson>}
+     * @memberof ICanFindOnlineBoilerplatePackages
+     */
+    discoverLatestOnlineBoilerplates(keywords?: string[], limit?: number): Promise<BoilerplatePackageJson[]>
+
+
+    /**
+     * Gets the package.json of the latest compatible boilerplate with the given package name
+     *
+     * @param {string} boilerplatePackageName
+     * @returns {Promise<BoilerplatePackageJson>}
+     * @memberof ICanFindOnlineBoilerplatePackages
+     */
+    latestCompatibleBoilerplate(boilerplatePackageName: string): Promise<BoilerplatePackageJson>
+
+    /**
+     * Finds boilerplate packages online and and returns the package.json objects of the latest versions of the boilerplates packages compatible with the common tooling major version
+     *
+     * @returns A list of compatible packages
+     * @returns {Promise<BoilerplatePackageJson>}
+     * @memberof ICanFindOnlineBoilerplatePackages
+     */
+    discoverLatestOnlineDolittleBoilerplates(): Promise<BoilerplatePackageJson[]>
```

Comment:
This method name was even more confusing; Dolittle.
Is this type taking on multiple responsibilities?
If its a case of differentiating @dolittle packages and others - could we do that without having specific methods and rather come up with a system that is extensible.